vmas/scenarios/navigation_obstacles.py

```diff
@@ -0,0 +1,412 @@
+                    x_bounds=self.x_bounds,
+                    y_bounds=self.y_bounds,
+                )
+                goal_poses.append(position.squeeze(1))
+                occupied_positions = torch.cat([occupied_positions, position], dim=1)
+
+        for i, agent in enumerate(self.world.agents):
+            if self.split_goals:
+                goal_index = int(i // self.agents_with_same_goal)
+            else:
+                goal_index = 0 if i < self.agents_with_same_goal else i
+
+            # set goal poses
+            agent.goal.set_pos(goal_poses[goal_index], batch_index=env_index)
+
+            if env_index is None:
+                agent.pos_shaping = (
+                    torch.linalg.vector_norm(
+                        agent.state.pos - agent.goal.state.pos,
+                        dim=1,
+                    )
+                    * self.pos_shaping_factor
+                )
+            else:
+                agent.pos_shaping[env_index] = (
+                    torch.linalg.vector_norm(
+                        agent.state.pos[env_index] - agent.goal.state.pos[env_index]
+                    )
+                    * self.pos_shaping_factor
+                )
+
+    def reward(self, agent: Agent):
+        is_first = agent == self.world.agents[0]
+
+        if is_first:
+            self.pos_rew[:] = 0
+            self.final_rew[:] = 0
+
+            for a in self.world.agents:
+                self.pos_rew += self.agent_reward(a)
+                a.agent_collision_rew[:] = 0
+
+            self.all_goal_reached = torch.all(
+                torch.stack([a.on_goal for a in self.world.agents], dim=-1), dim=-1
+            )
+
+            self.final_rew[self.all_goal_reached] = self.final_reward
+
+            # check for agent-agent collisions
+            for i, a in enumerate(self.world.agents):
+                for j, b in enumerate(self.world.agents):
+                    if i <= j:
+                        continue
+                    if self.world.collides(a, b):
+                        distance = self.world.get_distance(a, b)
+                        a.agent_collision_rew[
+                            distance <= self.min_collision_distance
+                        ] += self.agent_collision_penalty
+                        b.agent_collision_rew[
+                            distance <= self.min_collision_distance
+                        ] += self.agent_collision_penalty
+
+            # check for agent-obstacle collisions
+            for i, a in enumerate(self.world.agents):
+                for j, b in enumerate(self.world.landmarks[self.n_agents:]):
+                    if self.world.collides(a, b):
+                        distance = self.world.get_distance(a, b)
+                        a.agent_collision_rew[
+                            distance <= self.min_collision_distance
+                        ] += self.agent_collision_penalty
```

Comment:
Could you make this a separate variable? (e.g. one penalty for agent-agent collisions, an independent penalty for agent-obstacle collisions)

Reply:
sure! just to confirm, you mean having a `self.agent_agent_collision_penalty` and a `self.agent_obstacle_collision_penalty`